mlx/ops.cpp

```diff
@@ -2971,6 +2971,35 @@ array conv_general(
     input_dilation = std::vector<int>(spatial_dims, input_dilation_int);
   }
 
+  // Check for negative padding
+  bool has_neg_padding = false;
+  for (auto& pd : padding_lo) {
+    has_neg_padding = (pd < 0);
```

Comment:
If the first padding value is negative, but any subsequent value is positive, `has_neg_padding` will be set to `false` again and the negative padding won't be handled.